### DISCUSSION TITLE
Implementing the Canvas API

### DISCUSSION THREAD

**Discussion by mattlipman** [ucfopen/canvasapi#535] [Q&A] [answered]
Hi there,

As a Computer Science student, for my upcoming assignment, I am hoping to write a software application that leverages the Canvas API to take an assignment that my university sets and display it's due date and task-name within my own application. Does anyone know if such a process is possible via the Canvas API, and if so would anyone have any resources I could utilise to execute on the project. 

Thanks in advance.

FYI: The project I am writing now is written in flutter (Dart).

**Comment by bennettscience** [comment, marked as answer]
Yeah, you can definitely do this. I've never used Dart, so you'll need to just look through the Canvas API docs, particularly the Assignments endpoint. [There's a resource which lists all assignments for a user](https://canvas.instructure.com/doc/api/assignments.html#method.assignments_api.user_index), which should help you get started.

[You'll need to generate an API token in your account](https://community.canvaslms.com/t5/Student-Guide/How-do-I-manage-API-access-tokens-as-a-student/ta-p/273) to access any of the resources.

**Reply by mattlipman** [reply]
Thanks @bennettscience - I'll give this a go.
